src/command/dev/next/router/run.rs

```diff
@@ -120,53 +127,141 @@ impl RunRouter<state::LoadRemoteConfig> {
 }
 
 impl RunRouter<state::Run> {
-    pub async fn watch_for_changes<WriteF, Spawn>(
+    pub async fn run<Spawn>(
         self,
-        write_file_impl: WriteF,
         spawn: Spawn,
         temp_router_dir: &Utf8Path,
-    ) where
-        WriteF: WriteFile + Send + Clone + 'static,
+        studio_client_config: StudioClientConfig,
+        router_address: &RouterAddress,
+    ) -> Result<RunRouter<state::Watch>, RunRouterBinaryError>
+    where
         Spawn: Service<ExecCommandConfig, Response = Child> + Send + Clone + 'static,
         Spawn::Error: std::error::Error + Send + Sync,
         Spawn::Future: Send,
     {
         let config_path = temp_router_dir.join("config.yaml");
         let schema_path = temp_router_dir.join("supergraph.graphql");
```

Comment:
Maybe we should pass these in, since composition is producing these, and there should be a watcher listening for changes on the sources and writing to these paths?

Reply:
ahh, this sounds like it answers my question above; you're thinking we should pass the filepaths in rather than rely on the temp dir; I'll add a todo